db/schema_tables.cc

```diff
@@ -3035,10 +3035,6 @@ future<> maybe_update_legacy_secondary_index_mv_schema(service::migration_manage
     // format, where "token" is not marked as computed. Once we're sure that all indexes have their
     // columns marked as computed (because they were either created on a node that supports computed
     // columns or were fixed by this utility function), it's safe to remove this function altogether.
-    if (!db.features().cluster_supports_computed_columns()) {
-        return make_ready_future<>();
-    }
-
```

Comment:
Should we protect this with an even newer feature? e.g. cluster_computed_columns_schema_upgraded()?

So that we can skip the code below once it's been upgraded.

I'm not in favor of it myself, such games were just proven to be bad, I'm raising it for your consideration.

Reply:
I think there's no reason to - we're talking about a local schema upgrade, which is a glorified way of telling a local node that this column is computed. It will most likely not even get a chance to be propagated, but even if it does, the propagation code is already backward compatible and wouldn't send computed column information to nodes which do not understand it. So the check here was not only incorrect, but totally redundant.

Reply:
*just to make sure I'm not talking gibberish, I span a local 3-node cluster and tried to ruin it by removing the computed columns information from some nodes and restarting them, but everything worked properly

Reply:
Oh, I might have misunderstood the request. This code runs on startup only, so I think it's better for it to run redundantly than to introduce a new cluster feature.

Reply:
Hm, I also think that this code might run before the feature service has a chance to inquire other nodes about their cluster features... in which case this condition was as good as `if (false) { fix_stuff() }`.

Reply:
So, the original code had the condition reversed, as well as evaluated before its inputs were ready? Nice.

Reply:
Yeah, I'm particularly proud of that one. I think I'll print, laminate it and hang it on my wall.

Reply:
It's good that something like that can never get past reviewers.